sodym/named_dim_array_helper.py

```diff
@@ -0,0 +1,17 @@
+from .named_dim_arrays import NamedDimArray
+from .dimensions import Dimension, DimensionSet
+
+
+def named_dim_array_stack(named_dim_arrays: list[NamedDimArray], dimension: Dimension) -> NamedDimArray:
+    """Stack a list of NamedDimArray objects using a new dimension.
+    Like numpy.stack with axis=-1, but for `NamedDimArray`s.
+    Method can be applied to `NamedDimArray`s, `StockArray`s, `Parameter`s and `Flow`s.
+    """
+    named_dim_array0 = named_dim_arrays[0]
+    extended_dimensions = DimensionSet(dimensions=named_dim_array0.dims.dimensions+[dimension])
+    attrs = named_dim_array0.__dict__
+    attrs = {k: v for k, v in attrs.items() if k not in ['values', 'dims']}
```

Comment:
if we want to do a similar thing for the other operations, it would be good to have it as a method of NamedDimArray. And to exclude 'name' as well, that shouldn't be inherited, since the resulting array is quite often something else. And then instead of naming them explicitly, why not use `NamedDimArray.__fields__`?

I agree that `__dict__` is easier than `self.__class__.__fields__`, though.